module-4/backend/week_two.md

```diff
@@ -5,18 +5,31 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+  One difference between ES5 and ES6 is that in es5 you declare a variable using `var` whereas in es6 you can use `const` for unmutable values and `let` for mutable values. 
 2. What's the difference between asynchronous and synchronous JavaScript? 
+  Synchronous Javascript is where you have a singular call stack where all code is executed in order. Asynchronous Javascript is largely due to the fact that browsers give us additional resources to have a call stack for API requests. This way you have continue executing code while your program waits for the API request to resolve. 
 3. What are the four pillars of Object Oriented programming?
+  Polymorphism, Abstraction, Encapsulation, Inheritence
 4. What are some tools available in JavaScript to help you write object oriented code?
+  Object contrsuctor functions. These are functions that will return an object with properties defined in the function. You can pass arguments to these functions and they can be set to the properties of the objects. 
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+  Don't repeat yourself is a good concept to be aware of. Also, a sharp code smell is using set timeouts especially when you can chain promises together. 
```

Comment:
setTimeout is definitely a code smell some concept I would say are separation of responsibility such as DOM manipulation, data persistence etc